src/compound-option.cpp

```diff
@@ -50,35 +42,52 @@ void wf::config::update_compound_from_section(
                 // We have found a match.
                 // Find the suffix we should store values in.
                 std::string suffix = opt->get_name().substr(prefix.size());
-                if (!new_values.count(suffix) && (n > 0))
+                auto& tuple = new_values[suffix];
+
+                if (tuple.empty())
                 {
-                    // Skip entries which did not have their first value set,
-                    // because these will not be fully constructed in the end.
-                    continue;
+                    tuple.push_back(suffix);
+                    for (size_t i = 0; i < n; ++i)
```

Comment:
If I understand the code correctly, it will not work with all kinds of combinations. For example if I have `N` entries, but only the first and the last of them are specified in the config file, you will need a similar loop to fill in the middle entries with default values.

I think the following would be a simpler and more robust approach: first, iterate over all options and collect the suffixes. Then, iterate over all suffixes, iterate over all entries and check whether the corresponding options are present or the default value should be used.

Reply:
> you will need a similar loop to fill in the middle entries with default values.

There is such a loop below (lines 81-90)

> I think the following would be a simpler and more robust approach: first, iterate over all options and collect the suffixes.

I also thought of that too, but didn't want to rewrite it entirely. Will do that soon, thanks!